pennylane/ops/op_math/sprod.py

```diff
@@ -0,0 +1,219 @@
+
+    def terms(self):  # is this method necessary for this class?
+        r"""Representation of the operator as a linear combination of other operators.
+
+        .. math:: O = \sum_i c_i O_i
+
+        A ``TermsUndefinedError`` is raised if no representation by terms is defined.
+
+        .. seealso:: :meth:`~.Operator.compute_terms`
+
+        Returns:
+            tuple[list[tensor_like or float], list[.Operation]]: list of coefficients :math:`c_i`
+            and list of operations :math:`O_i`
+        """
+        return [self.scalar], [self.base]
+
+    @property
+    def is_hermitian(self):
+        """If the base operator is hermitian and the scalar is real,
+        then the scalar product operator is hermitian."""
+        return self.base.is_hermitian and (self.scalar.imag == 0)
+
+    def diagonalizing_gates(self):
+        r"""Sequence of gates that diagonalize the operator in the computational basis.
+
+        Given the eigendecomposition :math:`O = U \Sigma U^{\dagger}` where
+        :math:`\Sigma` is a diagonal matrix containing the eigenvalues,
+        the sequence of diagonalizing gates implements the unitary :math:`U`.
+
+        The diagonalizing gates rotate the state into the eigenbasis
+        of the operator.
+
+        A ``DiagGatesUndefinedError`` is raised if no representation by decomposition is defined.
+
+        .. seealso:: :meth:`~.Operator.compute_diagonalizing_gates`.
+
+        Returns:
+            list[.Operator] or None: a list of operators
+        """
+        return self.base.diagonalizing_gates()
+
+    def eigvals(self):
+        r"""Return the eigenvalues of the specified operator.
+
+        This method uses pre-stored eigenvalues for standard observables where
+        possible and stores the corresponding eigenvectors from the eigendecomposition.
+
+        Returns:
+            array: array containing the eigenvalues of the operator.
+        """
+        return self.scalar * self.base.eigvals()
+
+    def sparse_matrix(self, wire_order=None):
+        return _sprod(self.base.sparse_matrix(wire_order=wire_order), self.scalar)
+
+    def matrix(self, wire_order=None):
+        r"""Representation of the operator as a matrix in the computational basis.
+
+        If ``wire_order`` is provided, the numerical representation considers the position of the
+        operator's wires in the global wire order. Otherwise, the wire order defaults to the
+        operator's wires.
+
+        If the matrix depends on trainable parameters, the result
+        will be cast in the same autodifferentiation framework as the parameters.
+
+        A ``MatrixUndefinedError`` is raised if the base matrix representation has not been defined.
+
+        .. seealso:: :meth:`~.Operator.compute_matrix`
+
+        Args:
+            wire_order (Iterable): global wire order, must contain all wire labels from the operator's wires
+
+        Returns:
+            tensor_like: matrix representation
+        """
```

Comment:
Do we need to override the documentation?

Reply:
It currently doesn't show up in the read-the-docs. I suspect we do? I imagine that if we overwrite the function then it will by default overwrite the doc-string associated with the function? Not 100% sure